### PR TITLE
entity service unit test for relationship requests

### PR DESCRIPTION
**Pull request by manuelpolzhofer**
- added validator closure for better mocking
- added unit tests for `GetEntityByRelationship`

### CODE REVIEW

File: testingutils/documents/entityrelationship.go [hunk 1]
```diff
@@ -8,6 +10,7 @@ import (
 	"github.com/ethereum/go-ethereum/common/hexutil"
 )
 
+
```
Comment (charleenfei) [inline testingutils/documents/entityrelationship.go]: ?

File: documents/entity/service_test.go [hunk 2]
```diff
@@ -399,3 +482,113 @@ func TestService_calculateDataRoot(t *testing.T) {
...
+	docSrv := testingdocuments.MockService{}
+	mockedERSrv := &MockEntityRelationService{}
+	mockProcessor := &testingcommons.MockRequestProcessor{}
+
+	docSrv.On("GetCurrentVersion", eID).Return(entity, nil)
+	docSrv.On("Exists").Return(true).Once()
+	mockedERSrv.On("GetCurrentVersion", er.ID()).Return(er, nil)
+
+	fakeRoot, err := anchors.ToDocumentRoot(utils.RandomSlice(32))
+	nextId, err := anchors.ToAnchorID(entity.NextVersion())
```
Comment (charleenfei) [inline documents/entity/service_test.go]: assert.NoErr

File: documents/entity/service_test.go [hunk 3]
```diff
@@ -399,3 +482,113 @@ func TestService_calculateDataRoot(t *testing.T) {
...
+}
+
+func TestService_GetEntityByRelationship_latestInDB(t *testing.T) {
+	// prepare a service with mocked layers
+	ctxh, entity, er, idFactory, idService, repo := setupRelationshipTesting(t)
+
+	eID := entity.ID()
+	erID := er.ID()
+
+	// testcase: latest version in db
```
Comment (charleenfei) [inline documents/entity/service_test.go]: some testcases to add:
- if the document is not the latest version
- if the relationship is not found
- if the model is not an EntityRelationship

Comment (manuelpolzhofer) [reply documents/entity/service_test.go]: I will add them.

- notLatestVersion is already covered

**Pull request merged**